### PR TITLE
Starfield

### PR DESCRIPTION
**Pull request by Silarn**
Adds override plugin flag support, which coopts the old light flag and adds a new flag for light plugins.

We use a new game plugin property to determine which flag to check for light support.

### CODE REVIEW

File: src/record.h [hunk 1]
```diff
@@ -16,7 +16,9 @@ class Record
 public:
   enum EFlag {
     FLAG_MASTER = 0x00000001,
+    FLAG_LIGHT_ALTERNATE = 0x00000100,
     FLAG_LIGHT = 0x00000200,
+    FLAG_OVERRIDE = 0x00000200,
```
Comment (Al12rs) [inline src/record.h]: Are these two meant to be the same?

**Review by Al12rs — COMMENTED** [review]
I'm not too clear on what each flag represents, could we add comments to the enum explaining the differences?

File: src/record.h (hunk 1 above)
Comment (Silarn) [reply src/record.h]: Yes.

In SSE and FO4 the x200 flag means it's a light plugin. But in SF, x200 is a new 'override' plugin flag type and x100 is a light plugin.

Comment (Al12rs) [reply src/record.h]: Ah I see. Might be worth to add this information as comments

**Review by Al12rs — APPROVED** [review]

**Pull request merged**